### DISCUSSION TITLE
TIMEOUT error when ./run_tests -j4

### DISCUSSION THREAD

**Discussion by kmikityuk** [idaholab/moose#16788] [Q&A Getting Started] [answered]
Dear all,

I have successfully installed conda and moose according to the instructions for Linux. My OS: Red Hat Enterprise Linux Server 7.7 (Maipo). However, when run  `./run_test -j4` all not SKIPped tests failed with similar diagnostics, e.g.:
```
time_integrators/convergence.explicit_heun/level0 .................................................... RUNNING
time_integrators/convergence.explicit_heun/level0: Working Directory: /data/project/general/ans/COD/MOOSE/projects/moose/test/tests/time_integrators/convergence
time_integrators/convergence.explicit_heun/level0: Running command: /data/project/general/ans/COD/MOOSE/projects/moose/test/moose_test-opt -i explicit_convergence.i Executioner/TimeIntegrator/type=Heun Executioner/dt=0.00390625 Outputs/file_base=heun_0 --error --error-unused --error-override --no-gdb-backtrace
time_integrators/convergence.explicit_heun/level0: [mpiexec@merlin-l-001.psi.ch] match_arg (utils/args/args.c:163): unrecognized argument pmi_args
time_integrators/convergence.explicit_heun/level0: [mpiexec@merlin-l-001.psi.ch] HYDU_parse_array (utils/args/args.c:178): argument matching returned error
time_integrators/convergence.explicit_heun/level0: [mpiexec@merlin-l-001.psi.ch] parse_args (ui/mpich/utils.c:1642): error parsing input array
time_integrators/convergence.explicit_heun/level0: [mpiexec@merlin-l-001.psi.ch] HYD_uii_mpx_get_parameters (ui/mpich/utils.c:1694): unable to parse user arguments
time_integrators/convergence.explicit_heun/level0: [mpiexec@merlin-l-001.psi.ch] main (ui/mpich/mpiexec.c:148): error parsing parameters
time_integrators/convergence.explicit_heun/level0:
time_integrators/convergence.explicit_heun/level0 ................................ [FINISHED] FAILED (TIMEOUT)
```
As recommended I tried `./run_tests -i always_ok -p 2` and it finished successfully. I have also tried to increase the timeout limit, but without success.

Any ideas?

Best regards,

Konstantin

**Comment by GiudGiud** [comment, marked as answer]
Hi Konstantin

These look like MPI errors. Maybe the mpiexec run is not from the same distribution as the mpi library used for compiling.
What does 'which mpiexec' return?
In the moose conda environment, can you run a simple MPI hello world problem?

Best,
Guillaume

**Reply by kmikityuk** [reply]
Hi Guillaume,

`which mpiexec` correctly returns `~/miniconda3/envs/moose/bin/mpiexec`, but when I compiled and run `mpi_hello_world` code, I saw the same errors. So I checked my `bash_profile` and found the lines loading some modules for intel fortran that insludes openmpi. After commenting these lines, both `mpi_hello_world` and `run_test` work like a charm!

Thanks a lot for your advice!

Konstantin

**Reply by GiudGiud** [reply]
That's great news, thanks for letting us know. 
Guillaume
